lib/ap/amd64/sigsetjmp.s

```diff
@@ -0,0 +1,18 @@
+/*
+ * This file is part of the UCB release of Plan 9. It is subject to the license
+ * terms in the LICENSE file found in the top-level directory of this
+ * distribution and at http://akaros.cs.berkeley.edu/files/Plan9License. No
+ * part of the UCB release of Plan 9, including this file, may be copied,
+ * modified, propagated, or distributed except according to the terms contained
+ * in the LICENSE file.
+ */
+ 
+.globl sigsetjmp
```

Comment:
It's saving the mask. Equivalent to BSD:
```
	movl	%esi,88(%rdi)		/* 11; savemask */
	testl	%esi,%esi
```
I just aligned esi and then adding space for `sigset_t _psigblocked`.  Linux and musl does so. In fact after the jump, they push rdi and calls sigprocmask:
```
	pushq %rdi
	leaq 72(%rdi),%rdx
	xorl %esi,%esi
	movl $2,%edi
	call sigprocmask
	popq %rdi
```
but for us _psigblocked does the trick, APEX sigprocmask stores old mask in it. Also sigaction. And setjmp save all the regs. I thought in something similar for sigsuspend and sigpending which are dummies from old APE now and must be implemented.

You're in right, I must comment the code.